specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/blob.json

```diff
@@ -1240,6 +1240,13 @@
         "enabled": {
           "type": "boolean",
           "description": "Indicates whether change feed event logging is enabled for the Blob service."
+        },
+        "retentionInDays": {
+          "type": "integer",
+          "minimum": 1,
+          "maximum": 146000,
+          "format": "int32",
+          "description": "Indicates the duration of changeFeed retention in days. Minimum value is 1 day and maximum value is 146000 days (400 years). A null value indicates an infinite retention of the change feed."
```

Comment:
if customer doesn't specify the argument with only enabled True, is there a default value or it will get error?

Reply:
The default value is null, as mentioned in the description. There is no error thrown. The property is not marked required. It is nullable optional

Reply:
could I think it will be always enabled unless I disable it?

Reply:
Enabled changeFeed will generate logs, see [link](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-change-feed?tabs=azure-portal). Per my understanding, the value means how long the generated logs will be kept.
@HimanshuChhabra , could you clarify what will be retained in the description clearly?

Reply:
Yes the field indicates, on how long the feeds are to be retained. This description has come from the PM. I believe the line "Indicates the duration of changeFeed retention in days" makes it pretty clear.